test/functional/legacy/eval_spec.lua

```diff
@@ -495,6 +495,18 @@ describe('eval', function()
       ' abcE4b10-4\000abcE4b10-4-2')
   end)
 
+  it("setreg('0',getreg('a',1,1)) doesn't crash for unset register", function()
```

Comment:
well it is a functional test, `setreg(getreg())` is something a register plugin actually would do.  (and this is a real crash when using my own clipboard manager plugin).
But we should fix and test `setreg("0", v:_null_list)` as well, for sure (really I would like to just remove NULL list, we can use an interned fixed empty list for that purpose, but there is too many cases to check I guess so we need to support it for now)